src/EnergyPlus/PhotovoltaicThermalCollectors.hh

```diff
@@ -181,13 +233,36 @@ namespace PhotovoltaicThermalCollectors {
 
         void calculate(EnergyPlusData &state);
 
+        void calculateSimplePVT(EnergyPlusData &state);
+
+        void calculateBIPVT(EnergyPlusData &state);
+
+        void calculateBIPVTMaxHeatGain(
+            EnergyPlusData &state, Real64 tsp, std::string Mode, Real64 &bfr, Real64 &q, Real64 &tmixed, Real64 &ThEff, Real64 &tpv);
+
+        void solveLinSysBackSub(Real64 jj[9], Real64 f[3], Real64 (&y)[3]);
+
+        Real64 calc_k_taoalpha(Real64 theta, Real64 glass_thickness, Real64 refrac_index_glass, Real64 k_glass);
+
+        Real64 calc_taoalpha(Real64 theta, Real64 glass_thickness, Real64 refrac_index_glass, Real64 k_glass);
+
         void update(EnergyPlusData &state);
 
         void oneTimeInit(EnergyPlusData &state) override;
     };
 
     void GetPVTcollectorsInput(EnergyPlusData &state);
 
+    void GetPVTSimpleCollectorsInput(EnergyPlusData &state, int NumSimplePVTPerform, Array1D<SimplePVTModelStruct> &tmpSimplePVTperf);
+
+    void GetBIPVTCollectorsInput(EnergyPlusData &state, int NumBIPVTPerform, Array1D<BIPVTModelStruct> &tmpBIPVTperf);
+
+    void GetMainPVTInput(EnergyPlusData &state,
+                         int NumPVT,
+                         Array1D<PVTCollectorStruct> &PVT,
+                         Array1D<SimplePVTModelStruct> tmpSimplePVTperf,
```

Comment:
Pass arrays by `const &` not value.

Reply:
@amir, the array is being allocated in the method and this is why it was passed by reference?

Reply:
But `tmpSimplePVTperf` is being passed by value, which will copy the array.  Just pass it by const &.

Also, since we are already passing state everywhere, you are welcome to just put them on state itself like we do everywhere else in EnergyPlus.  Presumably they already live there anyway.